### PR TITLE
Find the requested hook only

### LINKED ISSUE #769: Simplify cookiecutter.hooks.find_hooks
We should rename `cookiecutter.hooks.find_hooks` to `find_hook(hook_name)` and explicitly look for the requested hook, instead of processing all the files in the hooks directory.

See https://github.com/audreyr/cookiecutter/pull/768/files/9a94484093ca23e9d55d42a53f096f67535b0b63#r68646614

### PR DESCRIPTION
**Pull request by michaeljoseph**
Fixes #769.

### CODE REVIEW

File: cookiecutter/hooks.py [hunk 1]
```diff
@@ -16,38 +16,43 @@
...
 
-        if basename in _HOOKS and not filename.endswith('~'):
-            hooks[basename] = os.path.abspath(os.path.join(hooks_dir, f))
-    return hooks
+        if (
+            basename in _HOOKS and
+            basename == hook_name and
+            not filename.endswith('~')
+        ):
+            return os.path.abspath(os.path.join(hooks_dir, f))
```
Comment (hackebrot) [inline cookiecutter/hooks.py]: For the sake of readability, I'd prefer to use the following:

```python
if basename not in _HOOKS or basename != hook_name:
    continue

if filename.endswith('~'):
    continue

return os.path.abspath(os.path.join(hooks_dir, f))
```

File: tests/test_hooks.py [hunk 2]
```diff
@@ -71,19 +71,31 @@ def test_find_hook(self):
...
-                'pre_gen_project': os.path.abspath('hooks/pre_gen_project.py'),
-                'post_gen_project': os.path.abspath(
-                    os.path.join('hooks', self.post_hook)
-                ),
-            }
-            assert expected == hooks.find_hooks()
+            assert (
+                os.path.abspath('hooks/pre_gen_project.py') ==
+                hooks.find_hook('pre_gen_project')
+            )
```
Comment (hackebrot) [inline tests/test_hooks.py]:
```python
expected_pre = os.path.abspath('hooks/pre_gen_project.py')
actual_pre = hooks.find_hook('pre_gen_project')
assert expected_pre == actual_pre
```

File: tests/test_hooks.py [hunk 3]
```diff
@@ -71,19 +71,31 @@ def test_find_hook(self):
...
+            assert (
+                os.path.abspath(os.path.join('hooks', self.post_hook)) ==
+                hooks.find_hook('post_gen_project')
+            )
```
Comment (hackebrot) [inline tests/test_hooks.py]: As above

**Comment by hackebrot** [comment]
Hey @michaeljoseph! Can we merge this?

**Comment by michaeljoseph** [comment]
@hackebrot yeah, I'm happy with this. :shipit:

**Pull request merged**